src/inc/gcinfotypes.h

```diff
@@ -255,24 +255,22 @@ struct GcStackSlot
 
 #elif defined(_TARGET_ARM64_)
 
-// ARM64TODO: these were copied from AMD64, and need to be reviewed
-
 #ifndef TARGET_POINTER_SIZE
 #define TARGET_POINTER_SIZE 8    // equal to sizeof(void*) and the managed pointer size in bytes for this target
 #endif
 #define NUM_NORM_CODE_OFFSETS_PER_CHUNK (64)
 #define NUM_NORM_CODE_OFFSETS_PER_CHUNK_LOG2 (6)
-#define NORMALIZE_STACK_SLOT(x) ((x)>>3)
+#define NORMALIZE_STACK_SLOT(x) ((x)>>3)   // GC Pointers word aligned
 #define DENORMALIZE_STACK_SLOT(x) ((x)<<3)
-#define NORMALIZE_CODE_LENGTH(x) (x)
-#define DENORMALIZE_CODE_LENGTH(x) (x)
-#define NORMALIZE_STACK_BASE_REGISTER(x) ((x) ^ 5)
-#define DENORMALIZE_STACK_BASE_REGISTER(x) ((x) ^ 5)
+#define NORMALIZE_CODE_LENGTH(x) ((x)>>2)   // All Instructions are 4 bytes long
+#define DENORMALIZE_CODE_LENGTH(x) ((x)<<2) 
+#define NORMALIZE_STACK_BASE_REGISTER(x) ((x)^29) // Encode Frame pointer X29 as zero
```

Comment:
NORMALIZE_STACK_BASE_REGISTER() encodes FP as 0 and SP as 2, to minimize the number of bits necessary to encode the base-register for reported stack slots. So, I believe this encoding is sufficient for ARM64.